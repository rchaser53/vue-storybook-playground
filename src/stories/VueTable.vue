<template>
  <div>
    <vuetable ref="vuetable"
      :data="hogeData" :apiMode="false"
      :fields="fields" pagination-path=""
      @vuetable:pagination-data="onPaginationData">
    <!-- api-url="https://vuetable.ratiw.net/api/users"   -->
    <!-- :muti-sort="true"  multi-sort-key="ctrl"  ##doen't work## -->
    </vuetable>
    <vuetable-pagination ref="pagination"
      @vuetable-pagination:change-page="onChangePage"
    ></vuetable-pagination>
  </div>
</template>

<script>
// import {Vuetable, VuetablePagination} from 'vuetable-2'
import {VuetablePagination} from 'vuetable-2'
import Vuetable from 'vuetable-2/src/components/Vuetable.vue'

export default {
  name: 'vue-table',
  data: function() {
    return {
      hogeData: [
        {
          name: 111,
          email: 222
        }
      ],
      fields: [
        {
          name: '__handle',   // only show the icon?
          dataClass: 'center aligned'
        },
        {
          name: 'name',
          sortField: 'name',
        }, {
          name: 'email',
          sortField: 'email',
          // direction: 'asc',      // doesn't work
          callback: 'nyan'
        }
      ]
    }
  },
  components: {
    Vuetable,
    VuetablePagination
  },
  methods: {
    onPaginationData (paginationData) {
      this.$refs.pagination.setPaginationData(paginationData)
    },
    onChangePage (page) {
      this.$refs.vuetable.changePage(page)
    },
    nyan(e) {
      // return value is the value in table
      return e;
    }
  }
}
</script>

<style>
  .vuetable-handle {
    width: 30px;
  }

  .handle-icon {
    display: block;
    width: 30px;
    height: 30px;
    background-color: green;
  }

  .pagination a.item {
    background-color: aqua;
    display: inline-block;
    text-align: center;
    margin: 3px;
    width: 30px;
  }
</style>