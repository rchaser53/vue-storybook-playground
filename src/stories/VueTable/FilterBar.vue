<template>
  <div class="filter-bar ui basic segment grid">
    <div class="ui form">
      <div class="inline field">
        <label>Search for:</label>
        <input type="text" v-model="filterText" class="three wide column" @keyup.enter="doFilter" placeholder="name, nickname, or email">
        <button class="ui primary button" @click="doFilter">Go</button>
        <button class="ui button" @click="resetFilter">Reset</button>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import Component from 'vue-class-component'

@Component
export default class FilterBar extends Vue {
  filterText = ''

  doFilter () {
    (this as any).$events.fire('filter-set', this.filterText)
  }
  resetFilter () {
    this.filterText = '';
    (this as any).$events.fire('filter-reset')
  }
}
</script>