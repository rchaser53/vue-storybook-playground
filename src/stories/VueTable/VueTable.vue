<template>
  <div>
    <filter-bar />
    <vuetable ref="vuetable"
      api-url="https://vuetable.ratiw.net/api/users"  
      :fields="fields" pagination-path="" track-by="name"
      detail-row-component="my-detail-row"
      @vuetable:cell-clicked="onCellClicked"
      @vuetable:pagination-data="onPaginationData" />

    <!-- :muti-sort="true"  multi-sort-key="ctrl"  ##doen't work## -->
    <vuetable-pagination ref="pagination"
      @vuetable-pagination:change-page="onChangePage" />
  </div>
</template>

<script>
import Vue from 'vue'
import { Vuetable, VuetablePagination } from 'vuetable-2'
import DetailRow from './VueTableDetailRow.vue'
import FilterBar from './FilterBar.vue'

Vue.component('filter-bar', FilterBar)
Vue.component('my-detail-row', DetailRow)

export default {
  // name: 'vue-table',
  mounted () {
    this.$on('filter-set', eventData => this.onFilterSet(eventData))
    this.$on('filter-reset', e => this.onFilterReset())
  },
  data: function() {
    return {
      hogeData: [
        {
          name: 111,
          email: 222
        }
      ],
      fields: [
        {
          name: '__handle',   // only show the icon?
          dataClass: 'center aligned'
        },
        {
          name: 'name',
          sortField: 'name',
        }, {
          name: 'email',
          sortField: 'email',
          // direction: 'asc',      // doesn't work
          callback: 'nyan'
        }
      ]
    }
  },
  components: {
    Vuetable,
    VuetablePagination
  },
  methods: {
    onPaginationData (paginationData) {
      this.$refs.pagination.setPaginationData(paginationData)
    },
    onChangePage (page) {
      this.$refs.vuetable.changePage(page)
    },
    onFilterSet (filterText) {
      this.moreParams.filter = filterText
      Vue.nextTick( () => this.$refs.vuetable.refresh() )
    },
    onFilterReset () {
      delete this.moreParams.filter
      Vue.nextTick( () => this.$refs.vuetable.refresh() )
    },
    nyan(e) {
      // return value is the value in table
      return e;
    },
    onCellClicked (data, field, event) {
      console.log('cellClicked: ', field.name, data.id)
      this.$refs.vuetable.toggleDetailRow(data.id)
    }
  }
}
</script>

<style>
  .vuetable-handle {
    width: 30px;
  }

  .handle-icon {
    display: block;
    width: 30px;
    height: 30px;
    background-color: green;
  }

  .pagination a.item {
    background-color: aqua;
    display: inline-block;
    text-align: center;
    margin: 3px;
    width: 30px;
  }
</style>