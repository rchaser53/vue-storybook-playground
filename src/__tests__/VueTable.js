import Vue from 'vue';
import VueTable from '../stories/VueTable/VueTable.vue';
import { createRenderer } from 'vue-server-renderer';
const renderer = createRenderer();

const render = (vm) => {
  return new Promise((resolve, reject) => {
    renderer.renderToString(vm, (err, str) => {
      if (err) {
        return reject(err);
      }
      resolve(str);
    });
  });
};

test('VueTableをrenderする', () => {
  const vm = new Vue({
  	el: document.createElement('div'),
  	components: {
  		'vue-table': VueTable,
  	},
  	render: (h) => {
  		return h('vue-table');
  	},
  });
  expect(vm.$el).toMatchSnapshot();
});
