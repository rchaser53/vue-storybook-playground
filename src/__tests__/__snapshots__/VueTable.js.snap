// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VueTableをrenderする 1`] = `
<div>
  <div
    class="filter-bar ui basic segment grid"
  >
    <div
      class="ui form"
    >
      <div
        class="inline field"
      >
        <label>
          Search for:
        </label>
         
        <input
          class="three wide column"
          placeholder="name, nickname, or email"
          type="text"
        />
         
        <button
          class="ui primary button"
        >
          Go
        </button>
         
        <button
          class="ui button"
        >
          Reset
        </button>
      </div>
    </div>
  </div>
   
  <table
    class="vuetable ui blue selectable celled stackable attached table"
    data-v-5cc42bfc=""
  >
    <thead
      data-v-5cc42bfc=""
    >
      <tr
        data-v-5cc42bfc=""
      />
    </thead>
     
    <tbody
      class="vuetable-body"
      data-v-5cc42bfc=""
    >
       
      <tr
        data-v-5cc42bfc=""
      >
        <td
          class="vuetable-empty-result"
          colspan="0"
          data-v-5cc42bfc=""
        >
          No Data Available
        </td>
      </tr>
       
    </tbody>
  </table>
   
  <div
    class="ui right floated pagination menu"
    style="display: none;"
  >
    <a
      class="btn-nav icon item"
    >
      <i
        class="angle double left icon"
      />
    </a>
     
    <a
      class="btn-nav icon item"
    >
      <i
        class="left chevron icon"
      />
    </a>
      
    <a
      class="btn-nav icon item"
    >
      <i
        class="right chevron icon"
      />
    </a>
     
    <a
      class="btn-nav icon item"
    >
      <i
        class="angle double right icon"
      />
    </a>
  </div>
</div>
`;
